components/pango_core/include/pangolin/maths/eigen_stack.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include <Eigen/Core>
+#include <pangolin/maths/eigen_concepts.h>
+
+namespace pangolin
+{
+
+template <typename TT>
+concept FloatingPointType = std::is_floating_point_v<TT>;
+
+template <typename TT>
+concept EigenOrFloatType = sophus::EigenType<TT> || FloatingPointType<TT>;
+
+template <typename Ts>
+struct DimTraits;
+
+template <FloatingPointType TT>
+struct DimTraits<TT> {
+  constexpr static int kNumRows = 1;
+  constexpr static int kNumCols = 1;
+};
+
+template <sophus::EigenType TT>
+struct DimTraits<TT> {
+  constexpr static int kNumRows = TT::RowsAtCompileTime;
+  constexpr static int kNumCols = TT::ColsAtCompileTime;
+};
+
+template <EigenOrFloatType... Ts>
```

Comment:
nice, I think I'll port this over to sophus - if this is okay

Reply:
sure! I've been wanting this for a while! Can finally write things like

Eigen::Vector3d pos3d = vstack(pos2d, 1.0);
...